### ISSUE TITLE
Generate Cypress E2E tests in TypeScript

### ISSUE THREAD

**Issue by Mikilll94** [vuejs/vue-cli#6241]
### What problem does this feature solve?
Cypress from version 4.4.0 supports writing E2E tests in TypeScript.

![Screenshot 2021-01-26 at 20 38 11](https://user-images.githubusercontent.com/17653648/105896097-ae9cf100-6016-11eb-8713-ad47af78ba8e.png)
https://docs.cypress.io/guides/references/changelog.html#4-4-0

Vue CLI should take this into account and create E2E tests in TS, when user chooses setup with TypeScript.

### What does the proposed API look like?
Generate Cypress E2E tests in TypeScript when the user selects setup with TypeScript.

**Comment by fredgan** [comment]
Agree. But it seems e2e tests are not limited to TS. You can also add e2e tests using Javascript.

**Comment by Mikilll94** [comment]
@fredgan 
That's why I wrote:

Generate Cypress E2E tests in TypeScript **when the user selects setup with TypeScript.**

For setup without TS, tests can be written in JS.
